src/neptune/internal/operation_processors/factory.py

```diff
@@ -53,13 +57,32 @@ def get_operation_processor(
     async_no_progress_threshold: float = ASYNC_NO_PROGRESS_THRESHOLD,
 ) -> OperationProcessor:
     if mode == Mode.ASYNC:
+        batch_size = int(os.environ.get(NEPTUNE_ASYNC_BATCH_SIZE) or "1000")
+
+        if os.getenv(NEPTUNE_ASYNC_PARTITIONS_NUMBER):
```

Comment:
I think we should add a new mode `Mode.ASYNC_PARTITIONED`. We can expose new mode - It will be mess in docs when we add too much env variables.